common/ledger/util/leveldbhelper/leveldb_provider_test.go

```diff
@@ -51,6 +52,53 @@ func TestIterator(t *testing.T) {
 	itr3 := db2.GetIterator(nil, nil)
 	defer itr3.Release()
 	checkItrResults(t, itr3, createTestKeys(0, 19), createTestValues("db2", 0, 19))
+
+	itr4 := db2.GetIterator(nil, nil)
+	defer itr4.Release()
+	require.True(t, itr4.Seek([]byte(createTestKey(10))))
+	require.Equal(t, []byte(createTestKey(10)), itr4.Key())
+	checkItrResults(t, itr4, createTestKeys(11, 19), createTestValues("db2", 11, 19))
+
+	itr5 := db2.GetIterator([]byte(createTestKey(11)), nil)
+	defer itr5.Release()
+	require.True(t, itr5.Seek([]byte(createTestKey(5))))
+	require.Equal(t, []byte(createTestKey(11)), itr5.Key())
+	checkItrResults(t, itr5, createTestKeys(12, 19), createTestValues("db2", 12, 19))
```

Comment:
Since all these tests are independent, it's better to run as a named sub-tests. For instance, this sub-test can be named as "seek-before-range"

Reply:
Overall, the `TestIterator` took <= 100 msecs. Given that we already run tests in parallel in CI build, I am not sure whether having parallel test here will help much. IMO, it just complicates the test as the seek needs to check for return `true` or `false`. If it is true, it needs to check the value at the current location. Further, I do use `First()`, `Prev()`, and `Last()` too. I am not convinced about this added complexity to the test.

Reply:
I think, I misread independent as a suggestion to use `t.Parallel` with the subtest. I think you meant to use only `t.Run` without `t.Parallel` when you mentioned independent.